service/static/js/rest_api.js

```diff
@@ -158,16 +158,34 @@ $(function () {
     // ****************************************
 
     $("#delete-btn").click(function () {
-
         var product_id = $("#recommendation_product_id").val();
         var related_product_id = $("#recommendation_related_product_id").val();
+        var type_id = $("#recommendation_type_id").val();
+        var status = $("#recommendation_status").val();
+        
+        if (product_id) {
+            var queryString = ""
+            if (related_product_id) {
+                queryString += "/" + related_product_id
+            } 
+            else if (type_id && status) {
+                queryString += "?type-id=" + type_id + "&status=" + status
+            } 
+            else if (type_id) {
+                queryString += "?type-id=" + type_id
+            } 
+            else if (status) {
+                queryString += "?status=" + status
+            }
+            else {
+                queryString += "/all"
+            }
 
-        if (product_id && related_product_id){
             var ajax = $.ajax({
                 type: "DELETE",
-                url: "/recommendations/" + product_id + "/" + related_product_id,
+                url: "/recommendations/" + product_id + queryString,
```

Comment:
This does not follow RESTful guidelines, if you want to use the queryString you have to include product_id inside the queryString as well

Reply:
On a second thought I think this is fine as we need atleast the product_id to delete something, but what happens when product id is not entered? It will just throw a Internal server error 500 right?

Reply:
I'm sorry I can put the product_id into the queryString.
 I thought it will be more clear at first, so I moved it out.
Now our server doesn't check the DELETE request without product id. Do we need to define a function to check it in the server?
Here in this website, I just throw out a message if the user doesn't input the product id.
Line 200 -202,
`else {
            flash_message("Please enter Product ID or Product ID with proper parameters")
        } `

Reply:
Oh sorry I did not see lines 200-202 before, everything looks fine you can merge it